web/html/src/utils/functions.ts

```diff
@@ -116,16 +116,34 @@ function capitalize(str: string): string {
 }
 
 function generatePassword(): string {
-  const length = Math.floor(Math.random() * 10) + 15;
-  // See https://stackoverflow.com/a/68617567/1470607
-  const charset = "abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ0123456789:-_";
-  const array = new Uint32Array(charset.length);
-  window.crypto.getRandomValues(array);
+  const length = 24; // fixed length for better security
+  const lowercase = "abcdefghijklmnopqrstuvwxyz";
+  const uppercase = "ABCDEFGHIJKLMNOPQRSTUVWXYZ";
+  const numbers = "0123456789";
+  const specials = ":-_!@#$%^&*()"; // extended set of special characters
+
+  // Helper function to get a random character from a given set
+  const allChars = lowercase + uppercase + numbers + specials;
+
+  // Ensure at least one character from each category
+  const getRandomChar = (charset) => charset[Math.floor(Math.random() * charset.length)];
```

Comment:
`Math.random()` is a pseudorandom number generator, I don't think we want to use it here. Similarly below in the sort function, I think this is not the way we want to approach this. The whole idea of using `getRandomValues()` in the original code is that we don't want to use `Math.random()`.